topi/tests/python_cpp/test_topi_transform.py

```diff
@@ -167,6 +167,55 @@ def check_device(device):
     for device in ["llvm", "nvptx", "cuda", "opencl", "metal", "rocm"]:
         check_device(device)
 
+def verify_take(src_shape, indices_shape, indices_val, axis=None):
+    dtype = "float32"
+    A = tvm.placeholder(shape=src_shape, dtype=dtype, name="A")
+    indices = tvm.placeholder(shape=indices_shape, dtype="int32", name="indices")
+    if axis is None:
+        out_tensor = topi.cpp.take(A, indices)
+    else:
+        out_tensor = topi.cpp.take(A, indices, axis)
+
+    def check_device(device):
+        ctx = tvm.context(device, 0)
+        if not ctx.exist:
+            print("Skip because %s is not enabled" % device)
+            return
+        print("Running on target: %s" % device)
+        with tvm.target.create(device):
+            s = topi.generic.schedule_injective(out_tensor)
+
+        foo = tvm.build(s, [A] + [indices] + [out_tensor] , device, name="take")
+        shape_size = 1
+        for i in range(len(src_shape)):
+            shape_size = shape_size * src_shape[i]
+        data_npy = np.arange(shape_size, dtype=dtype).reshape((src_shape))
+        indices_npy = np.zeros(indices_shape, dtype="int32")
+        if isinstance(indices_val, int):
+            indices_npy[0] = indices_val
+        else:
```

Comment:
The logic here seems unnecessarily complex. Why not just indices_npy = np.array(indices_val)?

When the indices is scaler, the output should be one dimension lower than the input.